lib/adapter/external/manga_kakalot.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Ewbscp.Adapter.External.MangaKakalot do
+  use Tesla
+
+  plug(Tesla.Middleware.BaseUrl, "https://ww7.mangakakalot.tv")
+
+  # returns the page with the search results (this would be a list of links to be presented to the user)
+  def search_manga(manga_title) do
+    case get("/search/" <> manga_title) do
+      {:ok, %{status: 200, body: body}} ->
+        body
+
+      {:error, reason} ->
+        {:error, reason}
+
+      _ ->
+        {:error, :unknown}
+    end
+  end
+
+  def get_manga_main_page(manga_main_page_uri) do
```

Comment:
Let's think about this: should the **external** adapter **receive** URIs?

Reply:
IMO, we should rethink that. 

Maybe isn't a `port` responsibility to know and send the URI for the adapter 🤔

Reply:
The adapter here is working as a **proxy**, only routing requests and returning its content without any transformation. We should have a model that represents the external content with loyalty, then the `port` will receive it and transform the correspondent **domain model** to which our business logic is applied. Right?

Reply:
As we talked earlier today, we're gonna merge this guy and then change all the structure as we want.